crux/apis.py

```diff
@@ -228,3 +228,31 @@ def set_datasets_provenance(self, provenance):
             "POST", ["datasets", "provenance"], headers=headers, json=provenance
         )
         return response.json()
+
+
+    def get_deliveries(self, dataset, start_date=None, end_date=None):
+
+        if isinstance(dataset, Dataset):
+            dataset_id = dataset.id
+        elif isinstance(dataset, str):
+            dataset_id = dataset
+        else:
+            raise TypeError("Please enter validate data type for dataset")
+
+        headers = Headers({"accept": "application/json"})
+
+        params = {}
+        params["start_date"] = start_date
+        params["end_date"] = end_date
+
+        deliveries = self.api_client.api_call(
+            "GET", ["deliveries", dataset_id, "ids"], headers=headers, params=params, model=Delivery
+        )
+
+        # Explictly setting datset_id property as the repsonse from /deliveries/dataset_id/ids only responds with delivery ids.
```

Comment:
* Explictly setting datset_id property as the repsonse from `/deliveries/dataset_id/ids` only responds with delivery ids.
* Fetching data endpoint requires dataset_id, for eg: `/deliveries/dataset_id/delivery_id/data`
* If this needs to be avoided then direct delivery endpoints can be used. for eg: `/deliveries/delivery_id/data` instead of the above.

Reply:
Thanks for it, added the following

- Added yield support
- Removed model from `api_call` and did explicit cast.